wordpress-mobile/WordPress-iOS
# Removes verticals and site info steps from site creation

Fixes #13666

##### Change log:
 - Update `SiteCreationWizardLauncher.swift`, remove `verticalStep` and `siteStep` from steps
- Update `SiteCreationWizardLauncher.swift`, remove `verticalStep` and `siteInfoStep` properties
 - Update `SiteCreator.swift`, assign default values for `title` and `tagline` in `SiteCreationRequest`
 - Update `SiteCreator.swift`, remove `invalidSiteInformation` error
- Update `InlineErrorTableViewProvider.swift`, add `typealias CellSelectionHandler = ((IndexPath) -> ())`, moved from `VerticalsTableViewProvider.swift`
- Delete `SiteVerticalsPromptService.swift`, not used anymore
- Delete files in `ViewRelated/Site Creation/SiteInfo`: 
  - `SiteInformationStep.swift`, `SiteInformationWizardContent.swift`, `SiteInformationWizardContent.xib`
- Delete files in `ViewRelated/Site Creation/Verticals`: 
  - `NewVerticalCell.swif`t, `NewVerticalCell.xib`, `SiteVerticalPresenter.swift`, `VerticalsCell.swift`, `VerticalsCell.xib`, `VerticalsStep.swift`, `VerticalsTableViewProvider.swift`, `VerticalsWizardContent.swift`, `VerticalsWizardContent.xib`.
- Update Release Notes.

To test:
- build/run
- Go to `My Sites`
- tap add site to launch the site creation wizard
- Verify that after choosing the type of site, the wizard skips to the `choose domain name` page
- Choose a domain name and verify that the site is created.

PR submission checklist:
